CMakeLists.txt

```
@@ -48,7 +49,43 @@ install(
   DESTINATION share/${PROJECT_NAME})
 
 ament_export_targets(${PROJECT_NAME})
-ament_export_dependencies(ament_cmake_google_benchmark benchmark osrf_testing_tools_cpp)
+ament_export_dependencies(
+  ament_cmake_google_benchmark benchmark osrf_testing_tools_cpp)
+
+if(BUILD_TESTING)
+  find_package(ament_lint_auto REQUIRED)
+  ament_lint_auto_find_test_dependencies()
+
+  set(performance_test_fixture_DIR "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+  add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
+  include(performance_test_fixture-extras.cmake)
+
+  add_performance_test(
+    benchmark_malloc_realloc
+    test/benchmark/benchmark_malloc_realloc.cpp
+    TIMEOUT 60)
+
+  # Adding executable and linking done manually to avoid dependency on
+  # osrf_testing_tools_cpp
+  if(TARGET benchmark_malloc_realloc)
+    ament_add_google_benchmark_executable(
+      benchmark_malloc_realloc_no_memory_tools
+      test/benchmark/benchmark_malloc_realloc_no_memory_tools.cpp)
+    if(TARGET benchmark_malloc_realloc_no_memory_tools)
+      target_link_libraries(
+        benchmark_malloc_realloc_no_memory_tools
+        benchmark::benchmark)
+
+      get_target_property(ARG_SKIP_RETURN_CODE benchmark_malloc_realloc SKIP_RETURN_CODE)
+      if(ARG_SKIP_RETURN_CODE)
+        set(ARG_SKIP_TEST "SKIP_TEST")
+      endif()
+
+      ament_add_google_benchmark_test(
+          benchmark_malloc_realloc_no_memory_tools ${ARG_SKIP_TEST})
+    endif()
+  endif()
```

Since `ament_add_google_benchmark` is just a vanilla Google Benchmark test, it has nothing to do with `osrf_testing_tools_cpp`. So you can use the existing macros, and this whole block gets a lot more simple:
```suggestion
  # Bypassing performance_test_fixture to avoid LD_PRELOAD of
  # osrf_testing_tools_cpp
  set(_no_mem_tools_skip_arg)
  if(NOT TARGET benchmark_malloc_realloc)
    set(_no_mem_tools_skip_arg SKIP_TEST)
  endif()
  ament_add_google_benchmark(
    benchmark_malloc_realloc_no_memory_tools
    test/benchmark/benchmark_malloc_realloc_no_memory_tools.cpp
    ${_no_mem_tools_skip_arg})
```

---

I like the simpler logic, but I also added a clause to check if benchmark_malloc_realloc is being skipped, then this one should be too.